src/reanimated2/hook/Hooks.ts

```diff
@@ -15,4 +16,30 @@ export function useWorkletCallback<A extends unknown[], R>(
   return useCallback(fun, deps ?? []);
 }
 
+export function useReducedMotion(): boolean {
+  const [shouldReduceMotion, setShouldReduceMotion] = useState(false);
+
+  useEffect(() => {
+    const handleChange = (isReduceMotionEnabled: boolean): void => {
+      setShouldReduceMotion(isReduceMotionEnabled);
+    };
+
+    async function init(): Promise<void> {
+      handleChange(await AccessibilityInfo.isReduceMotionEnabled());
+    }
+    init();
```

Comment:
Why not just the following?
```suggestion
    AccessibilityInfo.isReduceMotionEnabled().then(setShouldReduceMotion);
```

Reply:
probably with await here